language/evm/move-to-yul/src/solidity_ty.rs

```diff
@@ -0,0 +1,543 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Representation of solidity types and related functions.
+//! TODO: struct and function type
+
+use anyhow::{anyhow, Context as AnyhowContext};
+use itertools::Itertools;
+use regex::Regex;
+use std::{fmt, fmt::Formatter};
+
+use move_model::{
+    model::FunctionEnv,
+    ty::{PrimitiveType, Type},
+};
+
+use crate::context::Context;
+
+/// Represents a Solidity Signature appearing in the callable attribute.
+#[derive(Debug, Clone)]
+pub struct SoliditySignature {
+    pub sig_name: String,
+    pub para_types: Vec<(SolidityType, SignatureDataLocation)>,
+    pub ret_types: Vec<(SolidityType, SignatureDataLocation)>,
+}
+
+impl fmt::Display for SoliditySignature {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "sig:{} ", self.sig_name)?;
+        for (ty, data_location) in &self.para_types {
```

Comment:
Would it be difficult to produce here the correct Solidity syntax?

Then it could be easier to translate a Move signature just into this type, and then use display on this type to produce the signature.

Reply:
I added a function to generate the solidity signature for selector in the code.